extensions/cornerstone-dicom-sr/src/utils/convertAnnotations.ts

```diff
@@ -0,0 +1,237 @@
+            },
+            RelationshipType: 'SELECTED FROM',
+            ValueType: 'IMAGE',
+            ConceptNameCodeSequence: {
+              CodeValue: '111040',
+              CodingSchemeDesignator: 'DCM',
+              CodeMeaning: 'Original Source',
+            },
+          },
+          PixelOriginInterpretation: 'VOLUME',
+          GraphicData: imagePoint,
+          GraphicType: 'POINT',
+        };
+        newlyFoundReferencedImages.push({
+          ReferencedSOPClassUID: SOPClassUID,
+          ReferencedSOPInstanceUID: SOPInstanceUID,
+        });
+        let contentSequence;
+        if (i === 0) {
+          measurementGroupContentSequence[Object.keys(measurementGroupContentSequence).length] = {
+            ConceptNameCodeSequence: SCOORD3DContentItems[0].ConceptNameCodeSequence,
+            ContentSequence: newSCOORDContentItem,
+            ValueType: 'NUM',
+          };
+          contentSequence = measurementGroupContentSequence;
+          SCOORD3DContentItems[0].ValueType = 'SCOORD';
+          SCOORD3DContentItems[0].ContentSequence = {
+            ...SCOORD3DContentItems[0].ContentSequence,
+            ReferencedSOPSequence: {
+              ReferencedSOPClassUID: SOPClassUID,
+              ReferencedSOPInstanceUID: SOPInstanceUID,
+            },
+          };
+        } else {
+          const measurementGroupClone = Object.assign({}, measurementGroupMatrix);
+          measurementGroupClone.ContentSequence = [...contentSequenceMatrix];
+          contentSequence = toArray(measurementGroupClone.ContentSequence);
+          contentSequence[Object.keys(contentSequence).length] = {
+            ConceptNameCodeSequence: SCOORD3DContentItems[0].ConceptNameCodeSequence,
+            ContentSequence: newSCOORDContentItem,
+            ValueType: 'NUM',
+          };
+          contentSequence[scoord3DIndex] = { ...SCOORD3DContentItemsClone[0] };
+          contentSequence[scoord3DIndex].ContentSequence = {
+            ...SCOORD3DContentItems[0].ContentSequence,
+            ReferencedSOPSequence: {
+              ReferencedSOPClassUID: SOPClassUID,
+              ReferencedSOPInstanceUID: SOPInstanceUID,
+            },
+          };
+          contentSequence[trackIDIndex] = {
+            ...trackingIdentifierContentItemsClone,
+            UID: uid(),
+          };
+
+          newlyGeneratedMeasurementGroups.push(measurementGroupClone);
+        }
+        // add TrackingIdentifier in the new measurement group
+        const TrackingIdentifierGroup = contentSequence.find(
+          contentItem => contentItem.ConceptNameCodeSequence.CodeMeaning === TRACKING_IDENTIFIER
+        );
+
+        const mappedTrackingIdentifier = `${cornerstoneTag}:${toolName}`;
+        TrackingIdentifierGroup.TextValue = mappedTrackingIdentifier;
+      }
+    }
+  }
+  return { newlyGeneratedMeasurementGroups, newlyFoundReferencedImages };
+}
+
+export function convertAnnotations(dataset, displaySetService) {
```

Comment:
Same here, not sure what convertAnnotation means here.

Reply:
This function will be removed from here and put in the IDC mode